hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -32,6 +34,7 @@ func NewPropertyDefinition(propertyName PropertyName, jsonName string, propertyT
 		propertyType: propertyType,
 		jsonName:     jsonName,
```

Comment:
I think `jsonName` is fully superseded by `tags` and should be removed. This improves flexibility as currently we can't say `omitempty`.

Reply:
I've removed the property (just put the jsonName parameter into the tags collection in the constructor for now) .

There are probably cleaner ways to do it (so that we can actually support `omitempty`) but maybe better left for a follow up PR given this ones already large scope? Take a look and let me know what you think.